chain-signatures/node/src/protocol/posit.rs

```diff
@@ -0,0 +1,177 @@
+                store,
+            },
+        );
+
+        PositAction::Propose
+    }
+
+    // TODO: make the resp of this synchronous when each of the protocol managers
+    // are their own individual tasks such that they can respond without the need
+    // of the main consensus loop.
+    /// Act on the posit action. This will map the action received to a corresponding
+    /// action to be sent back to the proposer. This will return a series of internal
+    /// actions the node should take.
+    pub fn act(
+        &mut self,
+        id: T,
+        from: Participant,
+        threshold: usize,
+        action: &PositAction,
+    ) -> PositInternalAction<S> {
+        match action {
+            PositAction::Propose => {
+                if self.posits.contains_key(&id) {
+                    tracing::warn!(
+                        ?id,
+                        "received a protocol posit for an id that we already proposed"
+                    );
+                    return PositInternalAction::Reply(PositAction::Reject);
+                }
+
+                // No further checks necessary, we can just accept the posit.
+                PositInternalAction::Reply(PositAction::Accept)
+            }
+            PositAction::Start(participants) => {
+                if self.posits.contains_key(&id) {
+                    tracing::warn!(?id, "received an invalid Start for a protocol we proposed");
+                    return PositInternalAction::Reply(PositAction::Reject);
+                }
+
+                PositInternalAction::StartProtocol(
+                    participants.to_vec(),
+                    Positor::Deliberator(from),
+                )
+            }
+            PositAction::Accept | PositAction::Reject => {
+                let mut entry = match self.posits.entry(id) {
+                    Entry::Occupied(counter) => counter,
+                    Entry::Vacant(_) => {
+                        tracing::warn!(
+                            ?id,
+                            ?action,
+                            "received an action for a protocol we did NOT propose"
+                        );
+                        return PositInternalAction::None;
+                    }
+                };
+
+                let counter = entry.get_mut();
+                if action.is_accept() {
+                    counter.accepts.insert(from);
+                } else {
+                    counter.rejects.insert(from);
+                }
+
+                let enough_votes = counter.accepts.len() >= threshold
+                    && counter.accepts.len() + counter.rejects.len() == counter.participants.len();
+                if !enough_votes {
+                    return PositInternalAction::None;
+                }
+
+                tracing::info!(?id, "received all Accepts, starting protocol");
+                let counter = entry.remove();
+                let participants = counter.accepts.iter().copied().collect();
```

Comment:
Would it make sense to check if we are in the list of participants and reject otherwise?

Reply:
yeah that would make sense. There could be a case in the future where our changes lead to where we send out a start on a node that's not in the list